### DISCUSSION TITLE
webpack v5 and output 'commonjs2' with externals can not work in webpack v4

### DISCUSSION THREAD

**Discussion by Koreyhan** [webpack/webpack#14039] [Q&A]
# Bug report

**What is the current behavior?**

I use webpack v5, output 'commonjs2', and use externals config.
The output can not work in webpack v4, webpack-dev-server v3.

**If the current behavior is a bug, please provide the steps to reproduce.**

minimal repro:  https://github.com/Koreyhan/webpack5-bug-test/tree/master
1. In lib dir, `npm run build`, then `npm link`
2. In proj dir, `npm link @koreyhan/webpack5-lib`, then `npm run dev`
3. will get error in browser:  `Uncaught TypeError: __webpack_require__ is not a function`

**What is the expected behavior?**
It should work

**Other relevant information:**
webpack version:  lib: 5.51.1, proj: 4.2.0
Node.js version: v10.16.0
Operating System: Mac OS 11.3.1 
Additional tools:

**Comment by Koreyhan** [comment]
some pics

lib config
![image](https://user-images.githubusercontent.com/16666880/130227815-fc330ea9-be4c-4355-87fa-fec7aebeb9bf.png)

lib package.json
![image](https://user-images.githubusercontent.com/16666880/130227914-967f8a75-9aa2-4a6e-8827-32c9b47b1613.png)


proj config
![image](https://user-images.githubusercontent.com/16666880/130227867-73276729-2eb6-477f-b309-52dd2300e155.png)

proj package.json
![image](https://user-images.githubusercontent.com/16666880/130227946-2bbede86-1c68-4920-b6f8-8559caa14746.png)

proj index.js
![image](https://user-images.githubusercontent.com/16666880/130228178-f8a71794-0b91-432b-9d40-e81325a52471.png)


error msg
![image](https://user-images.githubusercontent.com/16666880/130228127-cd5dda81-56ba-48ab-ad38-093aa5bebd67.png)

**Reply by alexander-akait** [reply]
webpack v4 is outdated, please update to webpack v5

**Reply by alexander-akait** [reply]
And yes, there are small breaking changes

**Reply by Koreyhan** [reply]
But I have a v4 project to maintain, Is there any other way to support it?

**Reply by Koreyhan** [reply]
@alexander-akait if i turn optimization.minimize false to true, it woked. Can I use this in a production environment.

**Reply by abc3660170** [reply]
I meet the same problem, The reason is __webpack_require__ been overwrite,
